From 2b61fb934eca163c8de56020c44f25adb08a283f Mon Sep 17 00:00:00 2001
From: Yu Yue <yuyue@mprc.pku.edu.cn>
Date: Fri, 28 Oct 2011 19:15:39 +0800
Subject: [PATCH 041/641] UniCore64: Add function setup_arch

Fixed Error:
calibrate.c:(.init.text+0xa3c): undefined reference to `setup_arch'

Signed-off-by: Yu Yue <yuyue@mprc.pku.edu.cn>
---
 arch/unicore64/kernel/Makefile |    2 +-
 arch/unicore64/kernel/setup.c  |   12 ++++++++++++
 2 files changed, 13 insertions(+), 1 deletion(-)
 create mode 100644 arch/unicore64/kernel/setup.c

diff --git a/arch/unicore64/kernel/Makefile b/arch/unicore64/kernel/Makefile
index 1705a90..51ec843 100644
--- a/arch/unicore64/kernel/Makefile
+++ b/arch/unicore64/kernel/Makefile
@@ -1,3 +1,3 @@
-obj-y			:= tlb.o ptrace.o
+obj-y			:= tlb.o ptrace.o setup.o
 head-y			:= head.o
 extra-y			:= $(head-y) vmlinux.lds
diff --git a/arch/unicore64/kernel/setup.c b/arch/unicore64/kernel/setup.c
new file mode 100644
index 0000000..052b47c
--- /dev/null
+++ b/arch/unicore64/kernel/setup.c
@@ -0,0 +1,12 @@
+#include <linux/kernel.h>
+#include <linux/init.h>
+
+/**
+ * setup_arch() - Architecture-specific boot-time initializations
+ * @cmdline_p:
+ */
+void __init setup_arch(char **cmdline_p)
+{
+	/* FIXME */
+	BUG();
+}
-- 
1.7.9.5

